src/git.ts

```diff
@@ -52,6 +53,32 @@ export class Git {
     }
   }
 
+  public async clone(pwd: string, owner: string, repo:string) {
+    const { exitCode } = await this.git(
+        "clone",
+        [`https://x-access-token:${this.token}@github.com/${owner}/${repo}.git`],
+        pwd,
+    );
+    if (exitCode !== 0) {
+      throw new Error(
+          `'git clone ${owner}/${repo}' failed with exit code ${exitCode}`,
+      );
+    }
+  }
+
+  public async remoteAdd(pwd: string, source: string, owner: string, repo:string) {
+    const { exitCode } = await this.git(
+        "remote",
+        ["add", source, `https://x-access-token:${this.token}@github.com/${owner}/${repo}.git`],
```

Comment:
❓ Can you share where you got the `x-access-token:` prefix from? I can only find [other approaches](https://stackoverflow.com/a/66156992).

Reply:
I removed passing the token to `git.ts`. Instead, the token can be passed to the checkout step.

```yml
- uses: actions/checkout@v4
  with:
    token: ${{ secrets.MY_TOKEN }}
```